brainmatch/brainmatch.py

```diff
@@ -0,0 +1,326 @@
+
+    return project_features
+
+
+def compute_score(proj_features, contrib_data):
+
+    score = 0
+
+    # Assume that the integer indicating the skill is separated from the its
+    # meaning by a whitespace
+    contrib_git_skills = [int(s) for s in contrib_data[
+        experience_git_skills_field].split(" ") if s.isdigit()][0]
+
+    proj_git_skills = int(
+        proj_features[git_skills_label].split(underscore)[0]
+        if proj_features[git_skills_label] else -1)
+
+    # Split the dataframe strings into lists
+    contrib_experience_modality = \
+        [s.strip() for s in contrib_data[experience_modality_field].split(",")]
+    contrib_experience_programming = \
+        [s.strip()
+         for s in contrib_data[experience_programming_field].split(",")]
+    contrib_experience_tools = \
+        [s.strip() for s in contrib_data[experience_tools_field].split(",")]
+    contrib_experience_topic = \
+        [s.strip() for s in contrib_data[experience_topic_field].split(",")]
+
+    contrib_desired_modality = \
+        [s.strip() for s in contrib_data[desired_modality_field].split(",")]
+    contrib_desired_programming = \
+        [s.strip() for s in contrib_data[desired_programming_field].split(",")]
+    contrib_desired_tools = \
+        [s.strip() for s in contrib_data[desired_tools_field].split(",")]
+    contrib_desired_topic = \
+        [s.strip() for s in contrib_data[desired_topic_field].split(",")]
+
+    if contrib_git_skills == proj_git_skills:
+        score += 1
+
+    # Compute the scores corresponding to the contributor's experience
+    experience_modality_match = list(
+        set(proj_features[modality_label]) & set(contrib_experience_modality))
+    score += len(experience_modality_match)
+
+    experience_programming_match = list(
+        set(proj_features[programming_label]) &
+        set(contrib_experience_programming))
+    score += len(experience_programming_match)
+
+    experience_tools_match = list(
+        set(proj_features[tools_label]) & set(contrib_experience_tools))
+    score += len(experience_tools_match)
+
+    experience_topic_match = list(
+        set(proj_features[topic_label]) & set(contrib_experience_topic))
+    score += len(experience_topic_match)
+
+    # Compute the scores corresponding to the contributor's desired items
+    desired_modality_match = list(
+        set(proj_features[modality_label]) & set(contrib_desired_modality))
+    score += len(desired_modality_match)
+
+    desired_programming_match = list(
+        set(proj_features[programming_label]) &
+        set(contrib_desired_programming))
+    score += len(desired_programming_match)
+
+    desired_tools_match = list(
+        set(proj_features[tools_label]) & set(contrib_desired_tools))
+    score += len(desired_tools_match)
+
+    edesired_topic_match = list(
+        set(proj_features[topic_label]) & set(contrib_desired_topic))
+    score += len(edesired_topic_match)
```

Comment:
> total += len(proj_features[modality_label]) like you do for the contributor's experience.

I'd say that nothing is different between the modalities or the experience at this level, but anyways, maybe that was linked to the normalization of scores.

Otherwise, I think I see the issue you suggest: for an equal set of features, having two e.g. programming languages would be classified as a tighter project (higher score) than another one having just one. Good catch. The only way I can think of to normalize scores in this case would be by counting the number of project features/labels per category and normalizing the number of contributor matches against that denominator, then summing all scores and then dividing by the number of total features. I changed the PR accordingly in ca7af77. Does it seem OK to you?

Reply:
Done in 07290f7.